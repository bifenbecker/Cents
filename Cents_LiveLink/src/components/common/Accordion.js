import React, {useRef} from "react";
import {Box} from "rebass/styled-components";
import PropTypes from "prop-types";

import useToggle from "../../hooks/useToggle";

const Accordian = props => {
  const {label, children, labelStyles} = props;
  const childrenRef = useRef();

  const {isOpen, toggle} = useToggle();

  return (
    <>
      <Box onClick={toggle} {...labelStyles}>
        {isOpen ? "-" : "+"}
        &nbsp;
        {label}
      </Box>
      <Box
        {...styles.childrenWrapper}
        display={isOpen ? "block" : "hidden"}
        height={isOpen ? childrenRef.current.scrollHeight : 0}
        ref={childrenRef}
      >
        {children}
      </Box>
    </>
  );
};

const styles = {
  childrenWrapper: {
    height: 0,
    sx: {
      overflow: "hidden",
      transition: "all 0.3s ease-in-out",
    },
  },
};

Accordian.propTypes = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  children: PropTypes.node.isRequired,
  labelStyles: PropTypes.object,
};

Accordian.defaultProps = {
  labelStyles: {},
};

export default Accordian;
