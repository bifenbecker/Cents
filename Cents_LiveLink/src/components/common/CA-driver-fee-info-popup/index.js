import React, {useMemo} from "react";
import useWindowSize from "../../../hooks/useWindowSize";
import {DockModal} from "..";
import {Text, Flex, Button} from "rebass/styled-components";

const CADriverFeeInfoPopup = ({isOpen, close}) => {
  const [width, height] = useWindowSize();

  const getSize = useMemo(() => {
    if (height >= 568) {
      return 315;
    } else {
      return 0.75 * height;
    }
  }, [height]);

  const fontSize = width < 300 ? "14px" : width <= 340 ? "16px" : "18px";

  return (
    <DockModal isOpen={isOpen} provideBackOption={false} fixedSize size={getSize}>
      <Flex {...styles.mainWrapper}>
        <Flex {...styles.textContent}>
          <Text {...styles.boldText} fontSize={fontSize}>
            What is the CA Driver Fee?
          </Text>
          <Text {...styles.normalText}>
            Our on-demand delivery partner charges an additional fee each way for pickup
            and delivery in California. This fee helps cover benefits like health
            insurance and guaranteed earnings for on-demand pickup and delivery drivers.
          </Text>
          <Flex {...styles.footerWrapper}>
            <Button {...styles.closeButton} onClick={close}>
              Close
            </Button>
          </Flex>
        </Flex>
      </Flex>
    </DockModal>
  );
};

const styles = {
  mainWrapper: {
    sx: {
      height: "100%",
      flexDirection: "column",
      justifyContent: "space-between",
    },
    padding: "20px",
  },
  textContent: {
    flexDirection: "column",
    sx: {
      flex: 2,
    },
  },
  boldText: {
    color: "BLACK",
    fontWeight: 700,
    lineHeight: "21px",
  },
  normalText: {
    fontSize: "18px",
    color: "BLACK",
    fontFamily: "secondary",
    mt: "15px",
  },
  footerWrapper: {
    sx: {
      alignItems: "center",
      justifyContent: "center",
      width: "100%",
      height: "120px",
    },
  },
  closeButton: {
    sx: {
      backgroundColor: "CENTS_BLUE",
      width: "100%",
      borderRadius: 31,
      textTransform: "uppercase",
    },
    p: "20px",
  },
};

export default CADriverFeeInfoPopup;
