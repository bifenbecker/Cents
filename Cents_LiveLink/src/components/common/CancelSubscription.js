import React, {useMemo} from "react";
import PropTypes from "prop-types";
import {Flex, Button, Text} from "rebass/styled-components";
import {DockModal} from "./.";
import useWindowSize from "../../hooks/useWindowSize";

const CancelSubscription = props => {
  const [height] = useWindowSize();
  const {isOpen, toggle, dockProps, cancelSubscription} = props;

  const getSize = useMemo(() => {
    if (height >= 568) {
      return 250;
    } else {
      return 0.7 * height;
    }
  }, [height]);

  return (
    <DockModal
      {...dockProps}
      isOpen={isOpen}
      size={getSize}
      fixedSize
      provideBackOption={true}
      showExitIcon={true}
      header="Cancel Recurring Order"
      onBackClick={toggle}
    >
      <Flex {...styles.subscription}>
        <Text {...styles.subscription.subtext}>
          Are you sure you want to cancel this recurring order?
        </Text>
        <Flex {...styles.buttons}>
          <Button variant="thickOutline" {...styles.buttons.button} onClick={toggle}>
            DON'T CANCEL
          </Button>
          <Button
            variant="primary"
            {...styles.buttons.button}
            onClick={() => cancelSubscription(true)}
          >
            YES, CANCEL
          </Button>
        </Flex>
      </Flex>
    </DockModal>
  );
};

const styles = {
  subscription: {
    margin: "0px 30px",
    flexDirection: " column",
    alignItems: "center",

    exitIcon: {
      sx: {
        position: "absolute",
        top: "18px",
        left: "30px",
      },
    },

    subtext: {
      margin: "25px 0px",
      fontSize: "18px",
      fontFamily: "primary",
    },
  },
  buttons: {
    width: "100%",
    justifyContent: "space-between",
    alignItems: "center",
    my: "18px",

    button: {
      sx: {
        textTransform: "uppercase",
      },
      height: "60px",
      width: "calc(50% - 5px)",
      fontSize: ["12px", "14px", "16px"],
    },
  },
};

CancelSubscription.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
  dockProps: PropTypes.object,
  cancelSubscription: PropTypes.func.isRequired,
};

CancelSubscription.defaultProps = {
  isOpen: false,
  toggle: () => {},
  dockProps: {},
};

export default CancelSubscription;
