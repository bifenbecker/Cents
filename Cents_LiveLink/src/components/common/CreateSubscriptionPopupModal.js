import React, {useState, useEffect} from "react";
import PropTypes from "prop-types";
import {Flex, Text, Box, Image} from "rebass/styled-components";
import {SubscriptionModalImage} from "../../assets/images";
import {DockModal, Dropdown} from ".";
import useWindowSize from "../../hooks/useWindowSize";
import {INTERVAL_DISPLAY} from "../../constants/subscriptions";
import {
  COLORS,
  ORDER_DELIVERY_TYPES,
  ORDER_MESSAGES,
  SCHEDULE_BUTTONS,
  TEXT_ME_WHEN_READY,
} from "../../constants/order";
import {zIndex} from "styled-system";
import {IconBack, ExitIcon} from "../../assets/images";
import {Close} from "@material-ui/icons";
import {Button, Grid, IconButton, Typography} from "@material-ui/core";
import {SCHEDULE} from "constants/order";

const CreateSubscriptionPopupModal = (props) => {
  const {
    isOpen,
    toggle,
    onAddSubscription,
    onSkipSubscription,
    onScheduleDelivery,
    pickupTime,
    deliveryTime,
    discount,
  } = props;

  const closeModal = () => {
    toggle();
  };

  const [selectedInterval, setSelectedInterval] = useState(2);

  const handleAddSubscription = () => {
    onAddSubscription({interval: selectedInterval});
  };

  const getWeekListFromObject = Object.entries(INTERVAL_DISPLAY).map(
    ([interval, display]) => {
      return {label: display, value: Number(interval)};
    }
  );

  return (
    <div
      className={
        isOpen
          ? "recurring-choice-container open-from-bottom"
          : "recurring-choice-container"
      }
    >
      <div className="close-container">
        <IconButton>
          <Close fontSize="large" onClick={closeModal} alt="Dock Close" />
        </IconButton>
      </div>
      <Grid container className="title">
        <Typography variant="h1" component="h2">
          {SCHEDULE.recurringOrder}
        </Typography>
      </Grid>
      <Grid className="recurring-content">
        {discount ? (
          <Typography variant="h1" component="p" color="primary">
            {ORDER_MESSAGES.getSubscriptionDiscount(discount)}
          </Typography>
        ) : null}
        <Typography variant="h1" component="h2" color="p">
          {ORDER_MESSAGES.recurringCall}
        </Typography>
        <Grid>
          <Dropdown
            list={getWeekListFromObject}
            onListItemClick={({value}) => setSelectedInterval(value)}
            selectedListItem={selectedInterval}
          />
          <Text {...styles.wrapper.subHeaderText}>{SCHEDULE.pickup}</Text>
          <Text {...styles.wrapper.contentText}>{pickupTime}</Text>
          {deliveryTime === TEXT_ME_WHEN_READY ? (
            <Text
              {...styles.wrapper.customisecta}
              onClick={() => onScheduleDelivery(ORDER_DELIVERY_TYPES.pickup)}
            >
              {SCHEDULE_BUTTONS.edit}
            </Text>
          ) : null}
          <Text {...styles.wrapper.subHeaderText}>{SCHEDULE.delivery}</Text>
          <Text {...styles.wrapper.contentText}>{deliveryTime}</Text>
          {deliveryTime === TEXT_ME_WHEN_READY ? (
            <Text
              {...styles.wrapper.customisecta}
              onClick={() => onScheduleDelivery(ORDER_DELIVERY_TYPES.return)}
            >
              {SCHEDULE_BUTTONS.scheduleDeliveryNow}
            </Text>
          ) : (
            <Text
              {...styles.wrapper.customisecta}
              {...styles.wrapper.editSchedulingText}
              onClick={() => onScheduleDelivery(ORDER_DELIVERY_TYPES.pickup)}
            >
              {SCHEDULE_BUTTONS.editScheduling}
            </Text>
          )}
        </Grid>
        <Text {...styles.wrapper.note}>{SCHEDULE_BUTTONS.pickupReminder}</Text>
      </Grid>
      <Grid container className="recurring-controls">
        <Button
          variant="outlined"
          color="primary"
          className="schedule-button"
          onClick={onSkipSubscription}
        >
          {SCHEDULE_BUTTONS.notNow}
        </Button>
        <Button
          variant="contained"
          color="primary"
          className="schedule-button"
          onClick={handleAddSubscription}
        >
          {SCHEDULE_BUTTONS.apply}
        </Button>
      </Grid>
    </div>
  );
};

const styles = {
  wrapper: {
    height: "100%",
    flexDirection: "column",
    header: {
      marginLeft: "18px",
      marginRight: "18px",
      marginTop: "18px",
      marginBottom: "18px",
      fontSize: [16, 20],
    },
    headerblue: {
      color: "CENTS_BLUE",
      marginTop: "18px",
      marginLeft: "18px",
      marginRight: "18px",
      fontSize: [14, 16],
    },
    midContentWrapper: {
      display: "flex",
      flexDirection: "row",
      marginLeft: "18px",
      marginTop: "18px",
      marginBottom: "18px",
      justifyContent: "space-between",
    },
    midLeftContent: {
      width: "60%",
    },
    subHeaderText: {
      fontSize: "16px",
      marginTop: "24px",
    },
    contentText: {
      lineHeight: "24px",
      fontWeight: "100",
      fontSize: "16px",
    },
    customisecta: {
      color: "CENTS_BLUE",
      lineHeight: "23px",
      sx: {textDecoration: "underline", fontSize: 14},
    },
    editSchedulingText: {
      mt: "18px",
    },
    note: {
      margin: "30px 18px 0px 18px",
      color: "TEXT_GREY",
      fontSize: "14px",
    },
    buttoncontainer: {
      padding: "18px",
      display: "flex",
      flexDirection: "row",
      justifyContent: "space-between",
    },
    buttonoutline: {
      borderColor: "CENTS_BLUE",
      backgroundColor: "white",
      padding: "16px 16px",
      marginRight: "2%",
      textAlign: "center",
      color: "CENTS_BLUE",
      fontSize: ["14px", "16px"],
    },
    buttonsolid: {
      backgroundColor: "CENTS_BLUE",
      padding: "16px 24px",
      textAlign: "center",
      fontSize: ["14px", "16px"],
    },
  },
};

CreateSubscriptionPopupModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
  dockProps: PropTypes.object,
  pickupTime: PropTypes.string.isRequired,
  deliveryTime: PropTypes.string.isRequired,
  onAddSubscription: PropTypes.func.isRequired,
  onSkipSubscription: PropTypes.func.isRequired,
  onScheduleDelivery: PropTypes.func.isRequired,
  discount: PropTypes.number,
};

CreateSubscriptionPopupModal.defaultProps = {
  dockProps: {},
  discount: 0,
};

export default CreateSubscriptionPopupModal;
