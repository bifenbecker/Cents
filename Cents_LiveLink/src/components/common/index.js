export {default as Header} from "./Header";
export {default as Layout} from "./Layout";
export {default as Loader} from "./loader";
export {default as FullScreenModalForm} from "./FullScreenModalForm";
export {default as TextField} from "./TextField";
export {default as TextArea} from "./TextArea";
export {default as ToggleButton} from "./ToggleButton";
export {default as Accordion} from "./Accordion";
export {default as GlobalStyle} from "./GlobalStyle";
export {default as WithOrderStoreTheme} from "./WithOrderStoreTheme";
export {default as MaskedInput} from "./MaskedInput";
export {default as WithTheme} from "./WithTheme";
export {default as PillTabs} from "./PillTabs";
export {default as PillWithButton} from "./PillWithButton";
export {default as SimplePopover} from "./SimplePopover";
export {default as DockModal} from "./DockModal";
export {default as ToastError} from "./ToastError";
export {default as ScreenWrapper} from "./ScreenWrapper";
export {default as GoogleAddressSearchInput} from "./GoogleAddressSearchInput";
export {default as PickupDeliveryOrderSummary} from "./pickupDeliveryOrderSummary";
export {default as CADriverFeeInfoPopup} from "./CA-driver-fee-info-popup";
export {default as SmallFilledCircle} from "./SmallFilledCircle";
export {default as Dropdown} from "./Dropdown";
export {default as CancelSubscription} from "./CancelSubscription";
export {default as CurrentOrAllRecurringOrdersChoice} from "./CurrentOrAllRecurringOrdersChoice";
export {default as ReturnDeliveryConfirmationPopup} from "./return-delivery-confirmation-popup";
export {default as ServiceSelectionImageCard} from "./ServiceSelectionImageCard";
