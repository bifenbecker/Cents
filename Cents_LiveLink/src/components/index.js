// export {default as OrderSummary} from "./order-summary";
export {default as OrderSummary} from "./order-summary/new-index";
export {default as PageNotFound} from "./page-not-found";
export {default as VerifyAccount} from "./verify-account/VerifyAccount";
export {default as Landing} from "./landing";
export {default as OnlineOrder} from "./online-order";
export {default as MySubscriptions} from "./subscriptions";
