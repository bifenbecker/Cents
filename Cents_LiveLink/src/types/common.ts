export enum FetchingStatus {
  Initial,
  Pending,
  Fulfilled,
  Rejected,
}

export interface ISharedReducer {
  fetchingStatus: FetchingStatus;
  error: {
    text?: string;
    code?: string;
  };
}

export interface IGenericReducer<Type> extends ISharedReducer {
  data: Type;
}
