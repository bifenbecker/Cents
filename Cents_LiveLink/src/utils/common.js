import qs from "query-string";
import customerState, {initialCustomerState} from "../state/customer";

import {
  CUSTOMER_AUTH_TOKEN_KEY,
  CUSTOMER_KEY,
  BUSINESS_OWNER_AUTH_TOKEN_KEY,
} from "./config";

export const getQueryString = (search) => {
  return qs.parse(search?.split("?")[1]);
};

export const convertToQueryString = qs.stringify;

export const logoutCustomer = () => {
  localStorage.removeItem(CUSTOMER_AUTH_TOKEN_KEY);
  localStorage.removeItem(CUSTOMER_KEY);
  sessionStorage.removeItem(BUSINESS_OWNER_AUTH_TOKEN_KEY);
  customerState.set(initialCustomerState);
};

export const getDeliveryAcknowledgementKey = (orderId) => {
  return `liveLink:${orderId}-scheduledDeliveryAcknowledged`;
};

export const getDeliveryAfterProcessingAcknowledgementKey = (orderId) => {
  return `liveLink:${orderId}-scheduledDeliveryAfterProcessingAcknowledged`;
};

export const getSkipRecurringPickupAcknowledgementKey = (orderId) => {
  return `liveLink:${orderId}-skipRecurringPickupAcknowledged`;
};

export const setStringifiedLocalStorageData = (itemName, item) => {
  try {
    localStorage.setItem(itemName, JSON.stringify(item));
  } catch (error) {
    console.log(error);
  }
};

export const getParsedLocalStorageData = (itemName) => {
  try {
    return JSON.parse(localStorage.getItem(itemName)) || {};
  } catch (error) {
    return {};
  }
};

export const WEEKS = {
  WEEKLY: "Weekly",
  EVERY_2_WEEK: "Every 2 weeks",
  EVERY_3_WEEK: "Every 3 weeks",
  EVERY_4_WEEK: "Every 4 weeks",
};
