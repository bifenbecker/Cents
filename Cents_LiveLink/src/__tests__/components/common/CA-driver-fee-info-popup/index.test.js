import React from "react";
import {render, fireEvent, waitFor} from "@testing-library/react";

import {CADriverFeeInfoPopup} from "../../../../components/common";
import {act} from "react-dom/test-utils";

const renderCADriverInfoPopup = props => {
  const {getByText} = render(<CADriverFeeInfoPopup {...props} />);

  return {getByText};
};

describe("CADriverFeeInfoPopup Component", () => {
  const close = jest.fn();

  it("should match snapshot", () => {
    expect(render(<CADriverFeeInfoPopup isOpen close={close} />)).toMatchSnapshot();
  });

  it("should call close prop on clicking close button", async () => {
    const {getByText} = renderCADriverInfoPopup({
      isOpen: true,
      close,
    });

    const button = await getByText("Close");

    fireEvent.click(button);

    expect(close).toBeCalledTimes(1);
  });

  it("should set fontSize according to window width", async () => {
    const {getByText} = renderCADriverInfoPopup({
      isOpen: true,
      close,
    });
    const headerTxt = "What is the CA Driver Fee?";

    // width < 300
    act(() => {
      global.innerWidth = 250;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(headerTxt)).toHaveStyle({fontSize: "14px"}));

    // 300 < width <= 340
    act(() => {
      global.innerWidth = 340;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(headerTxt)).toHaveStyle({fontSize: "16px"}));

    // width > 340
    act(() => {
      global.innerWidth = 341;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(headerTxt)).toHaveStyle({fontSize: "18px"}));
  });
});
