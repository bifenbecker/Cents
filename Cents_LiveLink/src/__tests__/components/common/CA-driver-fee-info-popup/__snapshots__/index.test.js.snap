// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CADriverFeeInfoPopup Component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        style="position: fixed; width: 0px; height: 0px; top: 0px; left: 0px; z-index: 99999999;"
      >
        <div
          style="position: fixed; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; opacity: 1; transition: opacity 0.2s ease-out; background: white;"
        />
        <div
          style="position: fixed; z-index: 1; box-shadow: 0 0 4px rgba(0, 0, 0, 0.3); background: white; left: 25%; top: 0px; width: 50%; height: 768px; transition: left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out;"
        >
          <div
            style="position: absolute; z-index: 2; opacity: 0; top: -5px; height: 10px; left: 0px; width: 100%; cursor: row-resize;"
          />
          <div
            style="width: 100%; height: 100%; overflow: auto;"
          >
            <div
              class="sc-bczRLJ sc-gsnTZi iFtaFA hzPtvw"
            >
              <div
                class="sc-bczRLJ sc-gsnTZi iIJBxE hzPtvw"
              >
                <div
                  class="sc-bczRLJ cUdnpS"
                  color="BLACK"
                  font-size="18px"
                  font-weight="700"
                >
                  What is the CA Driver Fee?
                </div>
                <div
                  class="sc-bczRLJ egcaZu"
                  color="BLACK"
                  font-family="secondary"
                  font-size="18px"
                >
                  Our on-demand delivery partner charges an additional fee each way for pickup and delivery in California. This fee helps cover benefits like health insurance and guaranteed earnings for on-demand pickup and delivery drivers.
                </div>
                <div
                  class="sc-bczRLJ sc-gsnTZi eBiytq hzPtvw"
                >
                  <button
                    class="sc-bczRLJ fkzSit"
                  >
                    Close
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: fixed; width: 0px; height: 0px; top: 0px; left: 0px; z-index: 99999999;"
    >
      <div
        style="position: fixed; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; opacity: 1; transition: opacity 0.2s ease-out; background: white;"
      />
      <div
        style="position: fixed; z-index: 1; box-shadow: 0 0 4px rgba(0, 0, 0, 0.3); background: white; left: 25%; top: 0px; width: 50%; height: 768px; transition: left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out;"
      >
        <div
          style="position: absolute; z-index: 2; opacity: 0; top: -5px; height: 10px; left: 0px; width: 100%; cursor: row-resize;"
        />
        <div
          style="width: 100%; height: 100%; overflow: auto;"
        >
          <div
            class="sc-bczRLJ sc-gsnTZi iFtaFA hzPtvw"
          >
            <div
              class="sc-bczRLJ sc-gsnTZi iIJBxE hzPtvw"
            >
              <div
                class="sc-bczRLJ cUdnpS"
                color="BLACK"
                font-size="18px"
                font-weight="700"
              >
                What is the CA Driver Fee?
              </div>
              <div
                class="sc-bczRLJ egcaZu"
                color="BLACK"
                font-family="secondary"
                font-size="18px"
              >
                Our on-demand delivery partner charges an additional fee each way for pickup and delivery in California. This fee helps cover benefits like health insurance and guaranteed earnings for on-demand pickup and delivery drivers.
              </div>
              <div
                class="sc-bczRLJ sc-gsnTZi eBiytq hzPtvw"
              >
                <button
                  class="sc-bczRLJ fkzSit"
                >
                  Close
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
